ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activity-equation-answer/activityEquationAnswer.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
+import { Subscription } from 'rxjs';
+
+import { DecimalHelper } from '../../../../utility/decimalHelper';
+import { SubmissionManager } from '../../../../services/serviceAgents/submissionManager.service';
+import { DecimalAnswer } from '../../../../models/activity/decimalAnswer';
+import { ActivityEquationQuestionBlock } from '../../../../models/activity/activityEquationQuestionBlock';
+import { AnswerResponseEquation } from '../../../../models/activity/answerResponseEquation';
+
+@Component({
+  selector: 'ddp-activity-equation-answer',
+  template: `
+      <ddp-question-prompt [block]="block"></ddp-question-prompt>
+      <div class="equation-value">{{displayValue}}</div>
+  `
+})
+export class ActivityEquationAnswerComponent implements OnInit, OnDestroy {
+    @Input() block: ActivityEquationQuestionBlock;
+    private subscription: Subscription;
+
+    constructor(private submissionManager: SubmissionManager) {
+    }
+
+    ngOnInit(): void {
+        this.subscription = this.submissionManager.answerSubmissionResponse$.subscribe(response => {
+            this.updateEquationQuestions(response.equations);
+        });
+    }
+
+    ngOnDestroy(): void {
+        this.subscription.unsubscribe();
+    }
+
+    get displayValue(): string {
+        return this.block.answer && this.block.answer[0] ? this.formatValue(this.block.answer[0]) : null;
+    }
+
+    private updateEquationQuestions(equations: AnswerResponseEquation[] = []): void {
+        for (const equation of equations) {
+            // update only single equations (not equations which are children of a composite question)
+            if (this.block.stableId === equation.stableId && equation.values.length === 1) {
+                this.block.setAnswer(equation.values, false);
+            }
+        }
+    }
```

Comment:
This looks good! A code change suggestion to make this more compact: move the filtering (iteration through equations to find the one that matches this component) to the pipeline in the ngOnInit.

Reply:
Fixed in f7521bc1cc3665755cf7d381755d896d1804bf4d